query-engine/prisma/src/tests/type_mappings/test_api.rs

```diff
@@ -0,0 +1,211 @@
+use super::super::test_api::QueryEngine;
+use crate::context::PrismaContext;
+use quaint::prelude::Queryable;
+
+pub type TestResult = anyhow::Result<()>;
+
+pub struct TestApi {
+    provider: &'static str,
+    database_string: String,
+    is_pgbouncer: bool,
+}
+
+impl TestApi {
+    fn datasource(&self) -> String {
+        format!(
+            r#"
+                datasource my_db {{
+                    provider = "{provider}"
+                    url = "{url}"
+                }}
+            "#,
+            provider = self.provider,
+            url = self.database_string,
+        )
+    }
+
+    pub async fn execute(&self, sql: &str) -> anyhow::Result<()> {
+        let conn = quaint::single::Quaint::new(&self.database_string).await?;
+
+        conn.execute_raw(sql, &[]).await?;
+
+        Ok(())
+    }
+
+    pub async fn create_engine(&self) -> anyhow::Result<(DatamodelAssertions, QueryEngine)> {
```

Comment:
The name of this is a bit weird. So this creates a query engine and introspects the database, giving back a datamodel?

Reply:
good point, I'll change that

Reply:
renaming it to `introspect_and_start_query_engine`